### PR TITLE
OSX port

### PR DESCRIPTION
**Pull request by ukernel**
make source be able to compile (include debug programs and unit tests) on OS X
make vstart.sh work on OSX

### CODE REVIEW

**Comment by tchaikov** [comment]
i scheduled a rados qa run at http://149.202.175.39:8081/ubuntu-2015-08-04_13:28:43-rados-wip-osx-port---basic-openstack/ .

**Comment by tchaikov** [comment]
`Reviewed-by: Kefu Chai <kchai@redhat.com>` after the qa run

**Comment by tchaikov** [comment]
rebase on master to remove conflict.

**Comment by ukernel** [comment]
rebase on master to fix new compile errors

**Comment by ukernel** [comment]
the build error "libtool: link: cannot find the library `librados.la' or
unhandled argument `librados.la" was caused parallel jobs. no error when
using single thread  to build

**Comment by tchaikov** [comment]
reverted the LDADD to LDFLAGS chang and repushed.

**Comment by tchaikov** [comment]
rebase against master to remove uuid and gmtime commits for OSX. they are being taken care of already.

**Comment by tchaikov** [comment]
@ukernel 

the changes passed the rados suite. now waiting for @schafdog 's blessing. so i can add the "Signed-off-by" line for his commits.

**Comment by schafdog** [comment]
You have all the blessing from my side.

**Comment by schafdog** [comment]
New to ceph dev. Should I sign-off the commits?

**Comment by tchaikov** [comment]
hello @schafdog ! thanks for your great contribution.

> Should I sign-off the commits?

yes, we need the contributor's signed-off on his/her commits, see https://github.com/ceph/ceph/blob/master/SubmittingPatches#L62 . i was about to add them on behalf of you after getting your permission. but it would be ideal if you could do it for yours!

**Comment by tchaikov** [comment]
@schafdog probably in this case, it would be easier if i do it.

**Comment by schafdog** [comment]
I am off to work now, so I cannot do it until tonight. If you want to move ahead, you are still welcome to do it, and I will do it from now on.

**Comment by tchaikov** [comment]
i will do it then, have a good day!

**Comment by tchaikov** [comment]
added @schafdog 's signed-off-by to his commits. will merge once the build bot nods.

File: src/include/utime.h [hunk 1]
```diff
@@ -18,6 +18,15 @@
 #include <math.h>
 #include <sys/time.h>
 #include <time.h>
+
+#if defined(DARWIN)
+/* For some reason this definition is lost when compiling ceph *
+ * It works fine on a stand-alone file even with same clang    *
+ * parameters                                                  */
+
+time_t timegm(struct tm *);
```
Comment (tchaikov) [inline src/include/utime.h]: @schafdog seems we can dispense with this change as i am able to compile the tree without it.

and i checked "time.h" offered by apple. we should have `timegm()` declared if we include "time.h".

and we don't define `_ANSI_SOURCE` or `_POSIX_C_SOURCE` (i 

- put `#ifdefs #warning` in "test_crc32c.cc" to print messages if any of them are defined
- `make test/rgw/rgw_json_enc.o` without this change

to verify my theory).

**Pull request merged**